### PR TITLE
Homework3: Ivan Nikitsenka

### PR DESCRIPTION
**Pull request by vanich-nikita**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: 3.1 - FizzBuzz.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"""
+Напишите программу, которая печатает цифры от 1 до 100,
+но вместо чисел, кратных 3 пишет Fizz, вместо чисел кратныХ 5 пишет Buzz,
+а вместо чисел одновременно кратных и 3, и 5 - FizzBuzz
+"""
+
+for i in range(1, 101):
+    # для удобства вывода
+    if i == 52:
```
Comment (belovmd) [inline 3.1 - FizzBuzz.py]: what is it?

File: 3.7 - Chocolate bar.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+3. В третьем случае вводится переменная для посчета разломов,
+которая сравнивается с введенным значением.
+"""
+
+# 1.
+n = int(input("#1. Введите ширину: "))
+m = int(input("Введите длину: "))
+k = int(input("Введите площаль куска: "))
+
+if ((k % n == 0) or (k % m == 0)) and k < n * m:
```
Comment (belovmd) [inline 3.7 - Chocolate bar.py]: not k%n

File: 3.7 - Chocolate bar.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+res4 = k4 // n4
+m4 = m4 - res4
+t_res4 = k4 // n4 + (k4 % n4) // m4
+
+if (k4 % n4) % m4 == 0 and t == t_res4:
+    flag4 = True
+    print('Промежуточный результат: YES4')
+
+if flag3 or flag4:
+    print('Результат: YES')
```
Comment (belovmd) [inline 3.7 - Chocolate bar.py:119]: n = m = 10
t=10
k=1
it should be yes

File: 3.7 - Chocolate bar.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    # для проверки выполнения программы
+    print('Промежуточный результат: YES3')
+
+# разлом начинается с вертикали
+flag4 = False
+res4 = k4 // n4
+m4 = m4 - res4
+t_res4 = k4 // n4 + (k4 % n4) // m4
+
+if (k4 % n4) % m4 == 0 and t == t_res4:
```
Comment (belovmd) [inline 3.7 - Chocolate bar.py:114]: extremely difficult to understand all these magic names

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework3.md):
- 3
- 10
- 18
- 31
- 34

File: 3.1 - FizzBuzz.py (hunk 1 above)
Comment (vanich-nikita) [reply 3.1 - FizzBuzz.py]: deleted

File: 3.4 - Pairs of elements.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+Входные данные - строка из чисел, разделенная пробелами.
+Выходные данные - количество пар.
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+"""
+
+# lst = [1, 1, 2, 3, 6, 7, 88, 88, 88, 88, 1]
+lst = [1, 1, 1, 1]
+
+# lst = [int(s) for s in input().split()]
+# ctr = 0
```
Comment (belovmd) [inline 3.4 - Pairs of elements.py:14]: redundant comments

Comment (belovmd) [reply 3.4 - Pairs of elements.py:14]: not addressed previous issues

File: 3.4 - Pairs of elements.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+"""
+Первая единица образует пару с тремя другими единицами.
+Это подсчитывается через lst.count(x)-1. Количество пар находится
+для каждого элемента через for x in lst и складывается через sum().
+Сумма получается в два раза больше, т.к. каждая пара учитывается
+2 раза (например, первый элемент со вторым, а потом наоборот),
+поэтому добавлено деление на 2.
+"""
+print(sum(lst.count(x) - 1 for x in lst) // 2)
```
Comment (belovmd) [inline 3.4 - Pairs of elements.py:29]: can be significantly simpler

File: 3.5 - Unique items on the list.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+"""
+Дан список. Выведите те его элементы, которые встречаются в списке
+только один раз.
+Элементы нужно выводить в том порядке, в котором они встречаются в списке.
+"""
+
+lst = [1, 1, 2, 3, 3, 4, 4, 1, 0, 6, 7, 11]
+# final_list = []
+#
+# for i in lst:
```
Comment (belovmd) [inline 3.5 - Unique items on the list.py:10]: redundant comments

File: 3.5 - Unique items on the list.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+#     if lst.count(i) == 1:
+#         final_list.append(i)
+# print(final_list)
+
+
+def single_element(list_, result):
```
Comment (belovmd) [inline 3.5 - Unique items on the list.py:16]: O(n**2)

**Comment by belovmd** [comment]
redundant comments
not all comments addressed
7,5

**Pull request closed without merging**